float32 x
float32 y
float32 velX
float32 velZ